images/contestant/compilers.nix

```diff
@@ -0,0 +1,23 @@
+{ pkgs, lib, ... }:
+
+{
+  environment.systemPackages = with pkgs; [
+    # C
+    gcc
```

Comment:
Seeing this makes me nervous. Is there a way in this syntax that we can pin versions of compilers? The judgehosts are built on Ubuntu, and it's important the compiler versions match between what's in the judgehosts and what's on the contestant images.

Reply:
Yes, you can choose the package version if we want to

Reply:
See https://lazamar.co.uk/nix-versions/?channel=nixpkgs-24.05-darwin&package=gcc for an example